tests/integration/utils.ts

```diff
@@ -0,0 +1,220 @@
+import { AOProcess, AOS_MODULE_ID, AoClient } from '@ar.io/sdk';
+import AoLoader from '@permaweb/ao-loader';
+import fs from 'node:fs';
+import path from 'node:path';
+
+const __dirname = path.dirname(new URL(import.meta.url).pathname);
+export const TEST_AOS_ANT_WASM = fs.readFileSync(
+  path.join(__dirname, 'fixtures', `aos-ant-${AOS_MODULE_ID}.wasm`),
+);
+
+export const STUB_PROCESS_ID = 'process-id-'.padEnd(43, '1');
+export const STUB_AR_ADDRESS = 'arweave-address-'.padEnd(43, '1');
+export const STUB_ETH_ADDRESS = '0xFCAd0B19bB29D4674531d6f115237E16AfCE377c';
+export const STUB_ANT_REGISTRY_ID = 'ant-registry-'.padEnd(43, '1');
+
+export const AO_LOADER_HANDLER_ENV = {
+  Process: {
+    Id: STUB_AR_ADDRESS,
+    Owner: STUB_AR_ADDRESS,
+    Tags: [
+      { name: 'Authority', value: 'XXXXXX' },
+      { name: 'ANT-Registry-Id', value: STUB_ANT_REGISTRY_ID },
+    ],
+  },
+  Module: {
+    Id: ''.padEnd(43, '1'),
+    Tags: [{ name: 'Authority', value: 'YYYYYY' }],
+  },
+};
+
+export const AO_LOADER_OPTIONS = {
+  format: 'wasm32-unknown-emscripten-metering',
+  inputEncoding: 'JSON-1',
+  outputEncoding: 'JSON-1',
+  memoryLimit: '1073741824', // 1 GiB in bytes
+  computeLimit: (9e12).toString(),
+  extensions: [],
+};
+
+export const DEFAULT_HANDLE_OPTIONS = {
+  Id: ''.padEnd(43, '1'),
+  ['Block-Height']: '1',
+  // important to set the address so that that `Authority` check passes. Else the `isTrusted` with throw an error.
+  Owner: STUB_AR_ADDRESS,
+  Tags: [],
+  Module: 'ANT',
+  Target: STUB_AR_ADDRESS,
+  From: STUB_AR_ADDRESS,
+  Timestamp: Date.now(),
+  // for msg.reply
+  Reference: '1',
+  Data: '1234',
+};
+
+export type HandleFunction = Awaited<ReturnType<typeof AoLoader>>;
+
+/**
+ * @description Drop in replacement class representing the return type of the `connect` function from `@permaweb/aoconnect` (@type{AoClient})
+ * 
+ * This can be initialized and passed to AOProcess.ao
+ * 
+ * It satisfies our current use of AoClient.dryrun, AoClient.message, and AoClient.result by maintaining a `handle` function
+ * created with AoLoader, a `resultsCache`, and a txid-like `nonce`.
+ * 
+ * @example
+ * 
+ * ```ts
+ *  new AOProcess({
+      processId: 'ant-'.padEnd(43, '1'),
+      ao: (await LocalAO.init({
+        wasmModule: TEST_AOS_ANT_WASM,
+        aoLoaderOptions: AO_LOADER_OPTIONS,
+        handlerEnv: AO_LOADER_HANDLER_ENV,
+      })) as any,
+    });
+ * ```
+ */
+export class LocalAO implements Partial<AoClient> {
+  wasmModule: any;
+  handle: HandleFunction;
+  memory: ArrayBufferLike | null;
+
+  handlerEnv: typeof AO_LOADER_HANDLER_ENV;
+
+  nonce: string;
+  resultsCache: Map<string, Awaited<ReturnType<AoClient['result']>>> =
```

Comment:
i think it's fine, but curious to understand why you want to use a map here vs just "lastResult" or something similar. some concern this could grow unbounded in tests (may be better as an LRU). 

don't change it, just help me understand the thought process

Reply:
Agree with the LRU - also probably OK with lastResult, the thinking for me here was Map made it simple to imitate and integrate with how we current use AoClient in the AOProcess class.

Was also thinking of the other AoClient APIs like `results` - which i didn't implement because we don't currently use it - which returns paginated responses for all transactions for a process. So Map seemed like a good candidate.